api-server/api-server-common/src/storage/impls/postgres/mod.rs

```diff
@@ -57,7 +71,26 @@ impl TransactionalApiServerPostgresStorage {
             ))
         })?;
 
-        let result = Self { pool };
+        let (conn_tx, conn_rx) = tokio::sync::mpsc::unbounded_channel::<
+            PooledConnection<'static, PostgresConnectionManager<NoTls>>,
+        >();
+
+        let tx_dropper_joiner = tokio::task::spawn(async move {
+            let mut conn_rx = conn_rx;
+            while let Some(conn) = conn_rx.recv().await {
+                conn.batch_execute("ROLLBACK").await.unwrap_or_else(|e| {
```

Comment:
Looking at the documentation, maybe `batch()` rather than `batch_execute()` is better when only doing a single query?

Reply:
batch_execute() is what was used in postgres_async::Transaction. It seems the right thing to do. Not sure.